auth-api/src/auth_api/services/membership.py

```diff
@@ -315,14 +316,29 @@ def deactivate_membership(self, **kwargs):
     @staticmethod
     def _add_or_remove_group(model: MembershipModel):
         """Add or remove the user from/to account holders / product keycloak group."""
+        org_type_to_group_mapping = {
+            OrgType.MAXIMUS_STAFF.value: GROUP_MAXIMUS_STAFF,
+            OrgType.CONTACT_CENTRE_STAFF.value: GROUP_CONTACT_CENTRE_STAFF,
+        }
         if model.membership_status.id == Status.ACTIVE.value:
             KeycloakService.join_account_holders_group(model.user.keycloak_guid)
-        elif (
-            model.membership_status.id == Status.INACTIVE.value
-            and len(MembershipModel.find_orgs_for_user(model.user.id)) == 0
-        ):
+            if model.org.type_code in org_type_to_group_mapping:
+                KeycloakService.add_user_to_group(
+                    model.user.keycloak_guid, org_type_to_group_mapping[model.org.type_code]
+                )
```

Comment:
After adding them to these new external staff groups, we still need to:
1) check if they are part of the keycloak staff group and remove them
2) check if they have membership in the staff org and deactivate their membership

Reply:
I wouldn't put it "after adding them", I'd do that independently for the STAFF org

Reply:
Do we need a check to see if people are joining multiple staff orgs though?